src/TraceEvent/EventPipe/FixedBuffer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.Tracing.EventPipe
+{
+    internal sealed class FixedBuffer : MemoryManager<byte>
+    {
+        private IntPtr _pointer;
+        private int _length;
+
+        internal FixedBuffer(int length)
+        {
+            _pointer = Marshal.AllocHGlobal(length);
```

Comment:
Would NativeMemory be more appropriate here?

Reply:
Unforetunately TraceEvent targets .NET Standard 2.0 and NativeMemory was first supported in .NET 6 so it isn't available.